### ISSUE TITLE
Shader editor opening when clicking on node

### ISSUE THREAD

**Issue by Foyezes** [godotengine/godot#100152]
### Tested versions

4.4 dev5 and all godot 4 versions

### System information

Shader editor opening when clicking on node

### Issue description

Not sure if this is a bug or it's supposed to happen, but it can be annoying sometimes. When clicking on a mesh that has a shaderMaterial applied to it, and the material slot opened, clicking on that mesh will open the shader editor directly. This is unnecessary and is quite annoying if you're switching back and forth between editing shaders and working in the viewport.

### Steps to reproduce

Assign a shader material to any mesh

### Minimal reproduction project (MRP)

any new project

**Comment by matheusmdx** [comment]
Probably the same as #99772

Try to fold your shader property and see if stops:

![Image](https://github.com/user-attachments/assets/66cd8a6d-1c1d-41a3-92c3-f65b8c79ebe9)

**Comment by Foyezes** [comment]
> Probably the same as [#99772](https://github.com/godotengine/godot/issues/99772)
> 
> Try to fold your shader property and see if stops:
> 
> ![Image](https://github.com/user-attachments/assets/66cd8a6d-1c1d-41a3-92c3-f65b8c79ebe9)

Yes I knew that before. Is this intentional? There should be an option to turn off this behavior

**Comment by matheusmdx** [comment]
This is a regression, after unfocus the node that should auto-fold but that's broken right now.

**Comment by KoBeWi** [comment]
> after unfocus the node that should auto-fold but that's broken right now.

It's not that simple. Making it fold brings back #81481
Relevant code:
https://github.com/godotengine/godot/blob/0f95e9f8e665cedaa5aace5f6f86de2b4b5a8c60/editor/editor_properties.cpp#L3387-L3393

**Comment by Copperplate** [comment]
> Please note that focus will be stolen, even in the case where the user is inputing a search term in the inspector (for example: to find the material and close it ;))
> It is only possible to search after closing the material resource. Otherwise focus will continue to be stolen.

I just came across this too. Only the first letter of what you are searching for can be typed in the "Filter Properties" category of the Inspector, all subsequent letters will be hijacked and added to your script in the Shader Editor. **This is a very severe bug**. Thank you for looking into this.
